great_expectations_provider/operators/great_expectations.py

```diff
@@ -44,121 +78,404 @@ class GreatExpectationsOperator(BaseOperator):
             Callable[[CheckpointResult], None]
         ] = validation_failure_callback
         self.return_json_dict: bool = return_json_dict
+        self.use_open_lineage = use_open_lineage
+        self.is_dataframe = True if self.dataframe_to_validate is not None else False
+        self.datasource: Optional[Datasource] = None
+        self.batch_request: Optional[BatchRequestBase] = None
+
+        if self.is_dataframe and self.query_to_validate:
+            raise ValueError(
+                "Exactly one, or neither, of dataframe_to_validate or query_to_validate may be specified."
+            )
 
         # Check that only one of the arguments is passed to set a data context
-        if not bool(self.data_context_root_dir) ^ bool(self.data_context_config):
+        if not (bool(self.data_context_root_dir) ^ bool(self.data_context_config)):
             raise ValueError(
                 "Exactly one of data_context_root_dir or data_context_config must be specified."
             )
 
-        # Check that only one of the arguments is passed to set a checkpoint
-        if not bool(self.checkpoint_name) ^ bool(self.checkpoint_config):
+        if self.is_dataframe and self.conn_id:
+            raise ValueError(
+                "Exactly one, or neither, of dataframe_to_validate or conn_id may be specified. If neither is"
+                " specified, the data_context_root_dir is used to find the data source."
+            )
+
+        if self.query_to_validate and not self.conn_id:
+            raise ValueError(
+                "A conn_id must be specified when query_to_validate is specified."
+            )
+
+        # A data asset name is also used to determine if a runtime env will be used; if it is not passed in,
+        # then the data asset name is assumed to be configured in the data context passed in.
+        if (
+            self.is_dataframe or self.query_to_validate or self.conn_id
+        ) and not self.data_asset_name:
+            raise ValueError(
+                "A data_asset_name must be specified with a runtime_data_source or conn_id."
+            )
+
+        # Check that at most one of the arguments is passed to set a checkpoint
+        if self.checkpoint_name and self.checkpoint_config:
+            raise ValueError(
+                "Exactly one, or neither, of checkpoint_name or checkpoint_config may be specified. If neither is"
+                " specified, the default Checkpoint is used."
+            )
+
+        if (
+            not (self.checkpoint_name or self.checkpoint_config)
+            and not self.expectation_suite_name
+        ):
+            raise ValueError(
+                "An expectation_suite_name must be supplied if neither checkpoint_name nor checkpoint_config are."
+            )
+
+        if type(self.checkpoint_config) == CheckpointConfig:
+            self.checkpoint_config = self.checkpoint_config.to_json_dict()
+            self.checkpoint_config = {
```

Comment:
You are welcome to use `deep_filter_properties_iterable()` in `great_expectations/util.py`, if applicable.  Thanks.